### PR TITLE
whisper5: initial commit

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Tue Aug 30 11:47:53 UTC 2016_

**Comment by fjl** [comment]
please rename to whisper**v5**

File: whisper5/message.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	"golang.org/x/crypto/pbkdf2"
+)
+
+// Options specifies the exact way a message should be wrapped into an Envelope.
+type Options struct {
+	Topic TopicType
+	TTL   time.Duration
+	Src   *ecdsa.PrivateKey
+	Dst   *ecdsa.PublicKey
+	Key   []byte // must be 32 bytes. todo: review
```
Comment (zelig) [inline whisper5/message.go]: why not  use `common.Hash`?

Comment (gluk256) [inline whisper5/message.go]: Because the are logically unrelated.

**Comment by obscuren** [comment]
Even though you've closed the PR I'd like to comment on your commit messages.

Please use descriptive commits in the form of:

```
<package(s)>: short description

Long description with added functionality and changes
```

An example of bad descriptions are

* added some files
* initial commit
* fixed

Examples of proper short descriptions are

* shh: implemented whisper core
* shh: fixed sending fault in X

Of course messages would have a full description included as second paragraph. 

We also do not accept commits that "add files". Usually they belong to another commit. Neither do we accept commits that fixes something from a precious not-yet-merged commit; instead you should rebase and fix up the commit that has issue. 

And finally, having commits pushed to your PR that makes sense just to you and because you need them for save keeping during your development sprint are of course OK. As long as you rebase and clean up your commit history for the review.

**Pull request closed without merging**